tests/messages/test_frontend.py

```diff
@@ -761,6 +761,19 @@ def test_usage(self):
 pybabel: error: no valid command or option passed. try the -h/--help option for more information.
 """, sys.stderr.getvalue().lower())
 
+    def test_list_locales(self):
+        """
+        Test the command with the --list-locales arg.
+        """
+        result = self.cli.run(sys.argv + ['--list-locales'])
+        self.assertEqual(0, result)
+        output = sys.stdout.getvalue()
+        for line in output.splitlines():
+            self.assertRegex(
+                line,
+                r"[0-9A-Za-z_]{2,}\s+[0-9A-Za-z,&\\\-\.() ]+",
+            )
```

Comment:
I think we don't need to test this this strictly. How about something like
```suggestion
        output = sys.stdout.getvalue()
        # Known locale occurs in output
        assert 'fr_CH' in output
        assert 'French (Switzerland)' not in output
        # No bytes repr markers in output
        assert "\nb'" not in output
```